File: comandor/src/main/java/com/komandor/komandor/data/database/files/FileStorage.java
package com.komandor.komandor.data.database.files;

import com.komandor.komandor.data.api.model.response.MessageResponseModel;

import javax.inject.Inject;

import io.reactivex.Flowable;

public class FileStorage {
  
  FileDao fileDao;
  
  @Inject
  public FileStorage(FileDao fileDao) {
    this.fileDao = fileDao;
  }
  
  public long insertFile(MessageResponseModel message) {
    return fileDao.insertFile(new File(message));
  }
  
  public long insertFile(File file) {
    return fileDao.insertFile(file);
  }
  
  public Flowable<File> getFileByID(long id) {
    return fileDao.getFileByID(id);
  }
  
  public void updateFileID(long localFileID, String fileID) {
    fileDao.updateFileID(localFileID, fileID);
  }
  
  public void updateFilePathByFileID(String fileID, String filePath) {
    fileDao.updateFilePathByFileID(fileID, filePath);
  }
  
  public void updateFilePathByID(long id, String filePath) {
    fileDao.updateFilePathByID(id, filePath);
  }
}

File: comandor/src/main/java/com/komandor/komandor/di/FileStorage.kt
package com.komandor.komandor.di

import com.komandor.komandor.data.database.KomandorDatabase

import javax.inject.Inject

class FileStorage @Inject constructor(
    val db: KomandorDatabase
)  {
/*
    fun insertFile(messageResponse: MessageResponse): Long {
        return db.fileDao().insertFile(messageResponse.toFile())
    }

    fun insertFile(file: File): Long {
        return db.fileDao().insertFile(file)
    }

    fun getFileByID(id: Long): Flow<File?>? {
        return db.fileDao().getFileByID(id)
    }

    fun updateFileID(localFileID: Long, fileID: Long) {
        db.fileDao().updateFileID(localFileID, fileID)
    }

    fun updateFilePathByFileID(fileID: String?, filePath: String) {
        db.fileDao().updateFilePathByFileID(fileID, filePath)
    }

    fun updateFilePathByID(id: Long, filePath: String) {
        db.fileDao().updateFilePathByID(id, filePath)
    }

 */
}